src/k8s/cmd/k8s/k8s_bootstrap_test.go

```diff
@@ -0,0 +1,120 @@
+package k8s
+
+import (
+	"os"
+	"testing"
+
+	apiv1 "github.com/canonical/k8s/api/v1"
+	. "github.com/onsi/gomega"
+)
+
+var yamlConfigComplete = `
+components:
+  - network
+  - dns
+  - gateway
+  - ingress
+  - storage
+  - metrics-server
+cluster-cidr: "10.244.0.0/16"
+enable-rbac: true
+k8s-dqlite-port: 12379`
+
+var yamlConfigIncomplete = `
+cluster-cidr: "10.244.0.0/16"
+enable-rbac: true
+bananas: 5`
+
+func mustCreateTemporaryTestDirectory(t *testing.T) string {
+	// Create a temporary test directory to mock the snap
+	// <tempDir>
+	// 	└── init.yaml
+	t.Helper()
+
+	tempDir := t.TempDir()
+
+	err := os.MkdirAll(tempDir, 0777)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	return tempDir
+}
+
+func mustAddConfigToTestDir(t *testing.T, path string, data string) {
+	t.Helper()
+	// Create the init botstrap config file
+	err := os.WriteFile(path+"/init.yaml", []byte(data), 0644)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestGetConfigYaml(t *testing.T) {
+	t.Run("CompleteConfig", func(t *testing.T) {
+		g := NewWithT(t)
+
+		tempDir := mustCreateTemporaryTestDirectory(t)
```

Comment:
`t.TempDir()` also does everything you need
```suggestion
		tempDir := t.TempDir()
```